frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx

```diff
@@ -380,6 +380,28 @@ export default memo(function HeatMap({
     tissuesByName,
   ]);
 
+  useEffect(() => {
+    if (filteredTissueIds.length === 0) return;
+    setExpandedTissues(new Set(filteredTissueIds));
+    Object.entries(sortedCellTypesByTissueName).forEach(
+      ([tissue, cellTypes]) => {
+        if (filteredTissueIds.includes(tissuesByName[tissue].id)) {
+          cellTypes.forEach((cellType) => {
+            setDisplayedCellTypes(
```

Comment:
maybe need @seve 's input here, but `handleExpandCollapse()` seems to handle adding newly added tissue's cellTypes via `setDisplayedCellTypes()` here: https://github.com/chanzuckerberg/single-cell-data-portal/blob/b7024e7af98bd110f4ba8b80f9e1b2ac03d76908/frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx#L262

So perhaps potentially we could leverage `handleExpandCollapse()`?

Thank you!

Reply:
It could be possible, but we would have to modify that function to allow `tissueName` to be null. That function is also a bit more robust. Right now, this `useEffect` doesn't take filtered `cellTypes` into account.

Reply:
To be clear, I requested changes on the stance that there's a bug, I think there's still discussion to be had on whether we should use that function.